docs/resource-validation.md

```diff
@@ -0,0 +1,137 @@
+# Gateway API Resource Validation
+
+This document describes how NGINX Kubernetes Gateway (NKG) validates Gateway API resources.
+
+## Overview
+
+There are several reasons why NKG validates Gateway API resources:
+
+- *Robustness*, to gracefully handle invalid resources.
+- *Security*, to prevent malicious input from propagating to the NGINX configuration.
+- *Correctness*, to conform to the Gateway API specification for handling invalid resources.
+
+Ultimately, the goal is to ensure that NGINX continues to handle traffic even if invalid Gateway API resources were
+created.
+
+A Gateway API resource (a new resource or an update for the existing one) is validated by the following steps, some of
+which are provided by the Gateway API, while others are done by NKG:
+
+1. OpenAPI schema validation by the Kubernetes API server.
+2. Webhook validation by the Gateway API webhook.
+3. Webhook validation by NKG.
+4. Validation by NKG.
+
+To confirm that a resource is valid and accepted by NKG, check the `Accepted` condition in the resource status,
```

Comment:
"To confirm that a resource is valid and accepted by NKG, check that the `Accepted` condition in the resource status has a Status field of `True`.